specs/progress.md

```diff
@@ -0,0 +1,163 @@
+# Progress
+
+## Overview
+
+*Progress* is used to display the length of time a process will take (referred to as a **determinate** state) or to represent an unspecified wait time (referred to as an **indeterminate** state). *Progress* is typically visually represented by a circular or linear animation. When the `value` attribute is passed the state is **determinate**, otherwise it is **indeterminate**.
+
+### Use Cases
+
+The *progress* component should be used for an element that displays the progress status for a task that takes more than a few seconds or consists of several steps. Typical use cases include loading elements or status of a download. 
+
+- Jim is building a dashboard to track fullfilment for his online store he uses progress ring to visualize various metrics.
+
+- Susan is downloading a large collection of research music onto her harddrive. She sees a linear progress displaying the percentage complete for the currently downloading song and indeterminate linear progress for those songs in queue.
+
+- Fin is scrolling a large list of continuously loading data as he scrolls he sees a indeterminate progress ring to let him know that data is loading.
+
+### Types
+*Progress* can be visually represented in 2 ways.
+- Linear - A straight horizontal line. `fast-progress`
+- Circular - A cicular ring shape. `fast-progress-ring`
```

Comment:
are you saying this is a separate HTML element?

Reply:
Yes. We will have progress-base and then swap out templates and styles.

Reply:
Sorry I'm still confused. Do I, as an app author *using* this library, would write the following HTML?

```html
<!-- when I want a determinate progress -->
<fast-progress value="2" min="0" max="100"></fast-progress>

<!-- and this when I want a loading spinner-->
<fast-progress-ring></fast-progress-ring>
```